pkg/controller/cronjob/utils.go

```diff
@@ -321,3 +321,16 @@ func adoptJobs(sj *batchv2alpha1.CronJob, js []batchv1.Job, jc jobControlInterfa
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+func getCurrentTimeInZone(sj *batchv2alpha1.CronJob) time.Time {
+	if sj.Spec.Timezone == nil {
+		return time.Now()
+	}
+
+	loc, err := time.LoadLocation(*sj.Spec.Timezone)
+	if err != nil {
+		loc = time.Local
```

Comment:
This should also be an event. What I am wondering is if this is correct. time.Local means the local time this code runs ie. the time in the location the controller manager runs. Not sure we want that.